guacamole/src/main/webapp/app/client/controllers/clientController.js

```diff
@@ -776,6 +783,8 @@ angular.module('client').controller('clientController', ['$scope', '$routeParams
 
         // Client error
         else if (connectionState === ManagedClientState.ConnectionState.CLIENT_ERROR) {
+            // Stop intercepting window close
+            $window.removeEventListener('beforeunload', ctrlwlistener, false);
```

Comment:
Why repeatedly add, remove, add, remove, etc. the listener vs. checking the relevant state within the listener itself?

What about connections that are running in the background, as will occur if the user navigates back to the home screen or to a settings screen without disconnecting?

Reply:
I'm not much of a JS programmer.. couldn't find a way to test whether there was currently a connection.  If using connection groups, it would be ideal to be prompted even if at one of the settings screens, as they could lose the connection they had if it closes.

Reply:
The new commit has moved this to a check in the event handler.